queries.sql

```diff
@@ -16,8 +16,115 @@ SELECT * FROM animals WHERE name != 'Gabumon'; --Find all animals not named Gabu
 
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3; --Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
 
+/* 
+Inside a transaction update the animals table 
+by setting the species column to unspecified. 
+Verify that change was made. Then roll back 
+the change and verify that the species 
+columns went back to 
+the state before the transaction.
+*/
 BEGIN;
 UPDATE animals
 SET species = 'unspecified';
 
-ROLLBACK;
\ No newline at end of file
+ROLLBACK;
+
+/*
+Update the animals table by setting the 
+species column to digimon for all 
+animals that have a name ending in mon.
+*/
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+/*
+Update the animals table by setting 
+the species column to pokemon for all 
+animals that don't have species already set.
+*/
+BEGIN;
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+/*
+Inside a transaction delete all records in the animals table, 
+then roll back the transaction.
+*/
+BEGIN;
+
+DELETE FROM animals;
+ROLLBACK;
```

Comment:
Hello! . You're on the right track with using transactions and the `DELETE` statement. However, there are some minor adjustments needed to fully meet the project requirements. 🛠️

#### Highlights:
- You've initiated a transaction correctly with `BEGIN;`. 🎉
- Your `DELETE FROM animals;` query is spot on for removing all records from the `animals` table. 👍
- Good use of `ROLLBACK;` to undo the deletion operation. 👏

#### Areas for Improvement:
- Verification Missing: The project requirements specify that you should include `SELECT COUNT(*) FROM ANIMALS;` queries before and after the `ROLLBACK;` to verify that records are actually deleted and then restored. These lines are missing in your code.

Here's how you could complete the requirements:

```sql
BEGIN; -- start transaction

DELETE FROM animals; -- Delete all records

SELECT COUNT(*) FROM animals; -- 👈 Add this line to verify that records were deleted

ROLLBACK; -- Roll back the transaction

SELECT COUNT(*) FROM animals; -- 👈 Add this line to verify that records were restored
```

Please make these adjustments and re-submit your code. Looking forward to your next commit! 🚀

Best,
Temitope